src/Microsoft.DotNet.Build.Tasks/PackageFiles/OptionalTooling.targets

```diff
@@ -68,33 +73,16 @@
     <GenerateEncryptedNuGetConfig ConfigPath="$(OptionalRestoreConfigPath)"
                                   Sources="@(OptionalRestoreSource)" />
 
-    <EncryptedConfigNuGetRestore Inputs="$(OptionalToolingProjectJsonPath)"
+    <!-- restore the project.json file, if it is being used -->
+    <EncryptedConfigNuGetRestore Condition="'$(OptionalToolingProjectJsonPath)' != ''"
+                                 Inputs="$(OptionalToolingProjectJsonPath)"
                                  ConfigFile="$(OptionalRestoreConfigPath)"
                                  PackagesDir="$(PackagesDir)" />
-  </Target>
-
-  <!--
-    Restore optional tooling by generating the XML string of a NuGet.Config with plaintext
-    credentials, then injecting it into "dotnet restore" using /dev/stdin.
-    Not supported on Windows.
-  -->
-  <Target Name="RestoreOptionalToolingUnencrypted"
-          DependsOnTargets="CreateOptionalRestoreFeedItems;
-                            PrepareOptionalToolProjectJson"
-          Condition="'$(MSBuildRuntimeType)'=='core'">
-    <PropertyGroup>
-      <!-- Create a new restore command with no source parameters, only configfile. -->
-      <OptionalRestoreCommand>"$(DnuToolPath)"</OptionalRestoreCommand>
-      <OptionalRestoreCommand>$(OptionalRestoreCommand) restore</OptionalRestoreCommand>
-      <OptionalRestoreCommand Condition="'$(UseNuGetHttpCache)'!='true'">$(OptionalRestoreCommand) --no-cache</OptionalRestoreCommand>
-      <OptionalRestoreCommand>$(OptionalRestoreCommand) --packages "$(PackagesDir.TrimEnd('/\'.ToCharArray()))"</OptionalRestoreCommand>
-      <OptionalRestoreCommand>$(OptionalRestoreCommand) --configfile $(OptionalRestoreConfigPath)</OptionalRestoreCommand>
-    </PropertyGroup>
-
-    <Error Text="Restoring optional tooling with in-memory credentials is not supported on Windows. Use the desktop framework for the encrypted credential flow."
-           Condition="'$(RunningOnUnix)'!='true'" />
-
-    <Error Text="Restoring optional tooling with in-memory credentials is not yet implemented. No cross-platform optional tooling exists yet." />
+  
+    <!-- restore the MSBuild project file, if it is being used -->
+    <Exec Condition="Exists('$(OptionalToolingProjectPath)')" 
```

Comment:
I usually prefer != rather than Exists in places like that so that if it doesn't exist, I get an error at the first detectable point to make it obvious. Is there a situation where we legitimately get here without `OptionalToolingProjectPath` existing?

Reply:
Good point.  Switching to `!= ''`.  If the property is set, we should try using it.  It's an error case if the property is set to a non-existent file.